looks real

```
======================================================================
ERROR: test_force_consumption (corehq.apps.commtrack.tests.test_xml.CommTrackOTATest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/commtrack/tests/test_xml.py", line 156, in test_force_consumption
    balance_blocks = _get_ota_balance_blocks(self.domain, self.user)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/commtrack/tests/test_xml.py", line 617, in _get_ota_balance_blocks
    return extract_balance_xml(restore_config.get_payload().as_string())
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 498, in get_payload
    partial_response = provider.get_response(self.restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/providers.py", line 10, in get_response
    return get_case_payload(restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 26, in get_case_payload
    return sync_op.get_payload()
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 110, in get_payload
    self.restore_state.project, self.restore_state.stock_settings,
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 128, in __call__
    value = self.func( *args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 388, in stock_settings
    return self.project.commtrack_settings.get_ota_restore_settings()
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/commtrack/models.py", line 272, in get_ota_restore_settings
    if self.ota_restore_config.use_dynamic_product_list else []
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/products/models.py", line 158, in ids_by_domain
    return list(SQLProduct.objects.filter(domain=domain).product_ids)
TypeError: 'instancemethod' object is not iterable
----------------------------------------------------------------------
```

--- a/corehq/apps/commtrack/tests/test_products.py
+++ b/corehq/apps/commtrack/tests/test_products.py
@@ -26,7 +26,7 @@ class ProductsTest(CommTrackTest):
         )
 
         self.assertEqual(
-            len(Product.archived_by_domain(self.domain.name, wrap=False, include_archived=True)),
+            SQLProduct.objects.filter(domain=self.domain.name, is_archived=True).count(),
             1
         )
 

--- a/corehq/apps/products/models.py
+++ b/corehq/apps/products/models.py
@@ -155,15 +155,7 @@ class Product(Document):
 
     @classmethod
     def ids_by_domain(cls, domain):
-        """
-        Gets all product ids in a domain.
-        """
-        view_results = Product.get_db().view('commtrack/products',
-            startkey=[domain],
-            endkey=[domain, {}],
-            include_docs=False,
-        )
-        return [row['id'] for row in view_results]
+        return list(SQLProduct.objects.filter(domain=domain).product_ids())
 
     @classmethod
     def count_by_domain(cls, domain):

--- a/corehq/apps/products/models.py
+++ b/corehq/apps/products/models.py
@@ -147,41 +147,11 @@ class Product(Document):
             return cls.get(sql_product.product_id)
 
     @classmethod
-    def by_domain(cls, domain, wrap=True, include_archived=False, **kwargs):
-        """
-        Gets all products in a domain.
-
-        By default this filters out any archived products.
-        WARNING: this doesn't paginate correctly; it filters after the query
-        If you need pagination, use SQLProduct instead
-        """
-        kwargs.update(dict(
-            view_name='commtrack/products',
-            startkey=[domain],
-            endkey=[domain, {}],
-            include_docs=True
-        ))
-        if wrap:
-            products = Product.view(**kwargs)
-            if not include_archived:
-                return filter(lambda p: not p.is_archived, products)
-            else:
-                return products
-        else:
-            if not include_archived:
-                return [
-                    row["doc"] for row in Product.view(
-                        wrap_doc=False,
-                        **kwargs
-                    ) if not row["doc"].get('is_archived', False)
-                ]
-            else:
-                return [
-                    row["doc"] for row in Product.view(
-                        wrap_doc=False,
-                        **kwargs
-                    )
-                ]
+    def by_domain(cls, domain, wrap=True, include_archived=False):
+        queryset = SQLProduct.objects.filter(domain=domain)
+        if not include_archived:
+            queryset = queryset.filter(is_archived=False)
+        return list(queryset.couch_products(wrapped=wrap))
 
     @classmethod
     def archived_by_domain(cls, domain, wrap=True, **kwargs):

--- a/corehq/apps/products/models.py
+++ b/corehq/apps/products/models.py
@@ -154,14 +154,6 @@ class Product(Document):
         return list(queryset.couch_products(wrapped=wrap))
 
     @classmethod
-    def archived_by_domain(cls, domain, wrap=True, **kwargs):
-        products = cls.by_domain(domain, wrap, kwargs)
-        if wrap:
-            return filter(lambda p: p.is_archived, products)
-        else:
-            return [p for p in products if p.get('is_archived', False)]
-
-    @classmethod
     def ids_by_domain(cls, domain):
         """
         Gets all product ids in a domain.

